### DISCUSSION TITLE
Anyone facing this issue? Serialization of 'Symfony\Component\HttpFoundation\File\UploadedFile' is not allowed

### DISCUSSION THREAD

**Discussion by whousehomestay** [laravel/framework#46842] [Q&A]
I'm newbie here...
I don't know where is the exact problem, because the log only show anything about Queue-related-class. Here what log shows:

[2023-04-20 20:52:06] local.ERROR: Serialization of 'Symfony\Component\HttpFoundation\File\UploadedFile' is not allowed {"userId":27,"exception":"[object] (Exception(code: 0): Serialization of 'Symfony\\Component\\HttpFoundation\\File\\UploadedFile' is not allowed at /dir/vendor/laravel/serializable-closure/src/Serializers/Signed.php:68)
[stacktrace]
#0 /dir/vendor/laravel/serializable-closure/src/Serializers/Signed.php(68): serialize(Object(Laravel\\SerializableClosure\\Serializers\\Native))
#1 [internal function]: Laravel\\SerializableClosure\\Serializers\\Signed->__serialize()
#2 /dir/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(158): serialize(Object(Illuminate\\Queue\\CallQueuedClosure))
#3 /dir/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(127): Illuminate\\Queue\\Queue->createObjectPayload(Object(Illuminate\\Queue\\CallQueuedClosure), NULL)
#4 /dir/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(105): Illuminate\\Queue\\Queue->createPayloadArray(Object(Illuminate\\Queue\\CallQueuedClosure), NULL, '')
#5 /dir/vendor/laravel/framework/src/Illuminate/Queue/SyncQueue.php(38): Illuminate\\Queue\\Queue->createPayload(Object(Illuminate\\Queue\\CallQueuedClosure), NULL, '')
#6 /dir/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(253): Illuminate\\Queue\\SyncQueue->push(Object(Illuminate\\Queue\\CallQueuedClosure))
#7 /dir/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(229): Illuminate\\Bus\\Dispatcher->pushCommandToQueue(Object(Illuminate\\Queue\\SyncQueue), Object(Illuminate\\Queue\\CallQueuedClosure))
#8 /dir/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(95): Illuminate\\Bus\\Dispatcher->dispatchToQueue(Object(Illuminate\\Queue\\CallQueuedClosure))
#9 /dir/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(266): Illuminate\\Bus\\Dispatcher->dispatchSync(Object(Illuminate\\Queue\\CallQueuedClosure), NULL)
#10 /dir/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Bus\\Dispatcher->Illuminate\\Bus\\{closure}()
#11 /dir/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#12 /dir/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(81): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#13 /dir/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Object(Closure), Object(Closure))
#14 /dir/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Object(Closure), Array, NULL)
#15 /dir/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1250): Illuminate\\Container\\Container->call(Object(Closure))
#16 /dir/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(215): Illuminate\\Foundation\\Application->terminate()
#17 /dir/public/index.php(60): Illuminate\\Foundation\\Http\\Kernel->terminate(Object(Illuminate\\Http\\Request), Object(Illuminate\\Http\\RedirectResponse))
#18 {main}

Thank you...

**Comment by devajmeireles** [comment]
> I'm newbie here... I don't know where is the exact problem, because the log only show anything about Queue-related-class. Here what log shows:
> 
> [2023-04-20 20:52:06] local.ERROR: Serialization of 'Symfony\Component\HttpFoundation\File\UploadedFile' is not allowed {"userId":27,"exception":"[object] (Exception(code: 0): Serialization of 'Symfony\Component\HttpFoundation\File\UploadedFile' is not allowed at /dir/vendor/laravel/serializable-closure/src/Serializers/Signed.php:68) [stacktrace] #0 /dir/vendor/laravel/serializable-closure/src/Serializers/Signed.php(68): serialize(Object(Laravel\SerializableClosure\Serializers\Native)) #1 [internal function]: Laravel\SerializableClosure\Serializers\Signed->__serialize() #2 /dir/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(158): serialize(Object(Illuminate\Queue\CallQueuedClosure)) #3 /dir/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(127): Illuminate\Queue\Queue->createObjectPayload(Object(Illuminate\Queue\CallQueuedClosure), NULL) #4 /dir/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(105): Illuminate\Queue\Queue->createPayloadArray(Object(Illuminate\Queue\CallQueuedClosure), NULL, '') #5 /dir/vendor/laravel/framework/src/Illuminate/Queue/SyncQueue.php(38): Illuminate\Queue\Queue->createPayload(Object(Illuminate\Queue\CallQueuedClosure), NULL, '') #6 /dir/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(253): Illuminate\Queue\SyncQueue->push(Object(Illuminate\Queue\CallQueuedClosure)) #7 /dir/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(229): Illuminate\Bus\Dispatcher->pushCommandToQueue(Object(Illuminate\Queue\SyncQueue), Object(Illuminate\Queue\CallQueuedClosure)) #8 /dir/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(95): Illuminate\Bus\Dispatcher->dispatchToQueue(Object(Illuminate\Queue\CallQueuedClosure)) #9 /dir/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(266): Illuminate\Bus\Dispatcher->dispatchSync(Object(Illuminate\Queue\CallQueuedClosure), NULL) #10 /dir/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\Bus\Dispatcher->Illuminate\Bus\{closure}() #11 /dir/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}() #12 /dir/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(81): Illuminate\Container\Util::unwrapIfClosure(Object(Closure)) #13 /dir/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Object(Closure), Object(Closure)) #14 /dir/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Object(Closure), Array, NULL) #15 /dir/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1250): Illuminate\Container\Container->call(Object(Closure)) #16 /dir/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(215): Illuminate\Foundation\Application->terminate() #17 /dir/public/index.php(60): Illuminate\Foundation\Http\Kernel->terminate(Object(Illuminate\Http\Request), Object(Illuminate\Http\RedirectResponse)) #18 {main}
> 
> Thank you...

Have you tried queuing something related to the UploadedFile class?

**Reply by whousehomestay** [reply]
Owh yeah, thank you. So i use dispatch()->afterResponse() on upload file process and pass a $request variable which (maybe) contains uploadedFile class. This is not a problem before because it happens on production and just happened recently, i don't know exactly when. Is it prohibited now?

**Reply by whousehomestay** [reply]
So it happens after [v10.8.0](https://github.com/laravel/framework/releases/tag/v10.8.0) version i guess...

**Comment by loskov** [comment]
I have a similar problem with PDO:

```
Serialization of 'PDO' is not allowed {"exception":"[object] (Exception(code: 0): Serialization of 'PDO' is not allowed at /opt/project/vendor/laravel/serializable-closure/src/Serializers/Signed.php:68)
[stacktrace]
#0 /opt/project/vendor/laravel/serializable-closure/src/Serializers/Signed.php(68): serialize(Object(Laravel\\SerializableClosure\\Serializers\\Native))
#1 [internal function]: Laravel\\SerializableClosure\\Serializers\\Signed->__serialize()
#2 /opt/project/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(158): serialize(Object(Illuminate\\Queue\\CallQueuedClosure))
#3 /opt/project/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(127): Illuminate\\Queue\\Queue->createObjectPayload(Object(Illuminate\\Queue\\CallQueuedClosure), NULL)
#4 /opt/project/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(105): Illuminate\\Queue\\Queue->createPayloadArray(Object(Illuminate\\Queue\\CallQueuedClosure), NULL, '')
#5 /opt/project/vendor/laravel/framework/src/Illuminate/Queue/SyncQueue.php(38): Illuminate\\Queue\\Queue->createPayload(Object(Illuminate\\Queue\\CallQueuedClosure), NULL, '')
#6 /opt/project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(253): Illuminate\\Queue\\SyncQueue->push(Object(Illuminate\\Queue\\CallQueuedClosure))
#7 /opt/project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(229): Illuminate\\Bus\\Dispatcher->pushCommandToQueue(Object(Illuminate\\Queue\\SyncQueue), Object(Illuminate\\Queue\\CallQueuedClosure))
#8 /opt/project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(95): Illuminate\\Bus\\Dispatcher->dispatchToQueue(Object(Illuminate\\Queue\\CallQueuedClosure))
#9 /opt/project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(266): Illuminate\\Bus\\Dispatcher->dispatchSync(Object(Illuminate\\Queue\\CallQueuedClosure), NULL)
#10 /opt/project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Bus\\Dispatcher->Illuminate\\Bus\\{closure}()
#11 /opt/project/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#12 /opt/project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(81): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#13 /opt/project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Object(Closure), Object(Closure))
#14 /opt/project/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Object(Closure), Array, NULL)
#15 /opt/project/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1250): Illuminate\\Container\\Container->call(Object(Closure))
#16 /opt/project/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(215): Illuminate\\Foundation\\Application->terminate()
#17 /opt/project/public/index.php(55): Illuminate\\Foundation\\Http\\Kernel->terminate(Object(Illuminate\\Http\\Request), Object(Illuminate\\Http\\Response))
#18 {main}
"}
```

**Reply by cosmastech** [reply]
> I have a similar problem with PDO:
> 
> ```
> Serialization of 'PDO' is not allowed {"exception":"[object] (Exception(code: 0): Serialization of 'PDO' is not allowed at /opt/project/vendor/laravel/serializable-closure/src/Serializers/Signed.php:68)
> [stacktrace]
> #0 /opt/project/vendor/laravel/serializable-closure/src/Serializers/Signed.php(68): serialize(Object(Laravel\\SerializableClosure\\Serializers\\Native))
> #1 [internal function]: Laravel\\SerializableClosure\\Serializers\\Signed->__serialize()
> #2 /opt/project/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(158): serialize(Object(Illuminate\\Queue\\CallQueuedClosure))
> #3 /opt/project/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(127): Illuminate\\Queue\\Queue->createObjectPayload(Object(Illuminate\\Queue\\CallQueuedClosure), NULL)
> #4 /opt/project/vendor/laravel/framework/src/Illuminate/Queue/Queue.php(105): Illuminate\\Queue\\Queue->createPayloadArray(Object(Illuminate\\Queue\\CallQueuedClosure), NULL, '')
> #5 /opt/project/vendor/laravel/framework/src/Illuminate/Queue/SyncQueue.php(38): Illuminate\\Queue\\Queue->createPayload(Object(Illuminate\\Queue\\CallQueuedClosure), NULL, '')
> #6 /opt/project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(253): Illuminate\\Queue\\SyncQueue->push(Object(Illuminate\\Queue\\CallQueuedClosure))
> #7 /opt/project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(229): Illuminate\\Bus\\Dispatcher->pushCommandToQueue(Object(Illuminate\\Queue\\SyncQueue), Object(Illuminate\\Queue\\CallQueuedClosure))
> #8 /opt/project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(95): Illuminate\\Bus\\Dispatcher->dispatchToQueue(Object(Illuminate\\Queue\\CallQueuedClosure))
> #9 /opt/project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(266): Illuminate\\Bus\\Dispatcher->dispatchSync(Object(Illuminate\\Queue\\CallQueuedClosure), NULL)
> #10 /opt/project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Bus\\Dispatcher->Illuminate\\Bus\\{closure}()
> #11 /opt/project/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
> #12 /opt/project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(81): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
> #13 /opt/project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Object(Closure), Object(Closure))
> #14 /opt/project/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Object(Closure), Array, NULL)
> #15 /opt/project/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1250): Illuminate\\Container\\Container->call(Object(Closure))
> #16 /opt/project/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(215): Illuminate\\Foundation\\Application->terminate()
> #17 /opt/project/public/index.php(55): Illuminate\\Foundation\\Http\\Kernel->terminate(Object(Illuminate\\Http\\Request), Object(Illuminate\\Http\\Response))
> #18 {main}
> "}
> ```

Can you provide the code you're using that's triggering this?
